src/main/java/model/QualityIndicator.java

```diff
@@ -0,0 +1,114 @@
+package main.java.model;
+
+import java.util.ArrayList;
+
+/**
+ * A class that represents the indicators quality of the tools in comparison with iPlasma and PDM
+ */
+
+public class QualityIndicator {
+
+	private int PMDDCI, PMDDII, PMDADCI, PMDADII, iPlasmaDCI, iPlasmaDII, iPlasmaADCI, iPlasmaADII;
+	private ArrayList<ExcelRow> excelRows;
+	
+	/**
+	 * Creates a IndicatorQuality object based on:
+	 *
+	 * @param PMDDCI - DCI Indicator - number of faults correctly identified of PDM tool
+	 * @param PMDDII - DII Indicator - number of faults incorrectly indetified of PDM tool
+	 * @param PMDADCI ADCI Indicator - number of absence of faults correctly indetified of PDM tool
+	 * @param PMDADII - ADII Indicator - number of absence of faults incorrectly indetified of PDM tool
+	 * @param iPlasmaDCI - DCI Indicator - number of faults correctly identified of iPlasma tool
+	 * @param iPlasmaDII - DII Indicator - number of faults incorrectly indetified of iPlasma tool
+	 * @param iPlasmaADCI ADCI Indicator - number of absence of faults correctly indetified of iPlasma tool
+	 * @param iPlasmaADII - ADII Indicator - number of absence of faults incorrectly indetified of iPlasma tool
+	 * 
+	 * A IndicatorQuality object also compute the Quality Indicators 
+	 */
+	public QualityIndicator(ArrayList<ExcelRow> excelRows) {
+		PMDDCI = 0; 
+		PMDDII = 0; 
+		PMDADCI = 0; 
+		PMDADII = 0;
+		iPlasmaDCI = 0;
+		iPlasmaDII = 0;
+		iPlasmaADCI = 0;
+		iPlasmaADII = 0;
+		this.excelRows = excelRows;
+		
+		for (ExcelRow row : excelRows) {
+			if(row.isPMD() && row.isIs_long_method())
```

Comment:
we either need to document what this is doing or I must say it's not clear at all - we may need to refactor these methods, it's not clear what's happening here and what's being compared exactly

Reply:
Os metodos da classe ExcelRow para verificar se PMI = TRUE or FALSE (Lino). Não vejo como mudar isto.
if(PMD= TRUE && is_long_method=TRUE)  -> Acrescenta o valor do PMDDCI (da comparação com o PMD) É isto que está no código 

- DCI, quando o valor da coluna correspondente à ferramenta (PMI ou iPlasma) é TRUE e a coluna is_long_method também é TRUE
- DII, quando o valor da coluna correspondente à ferramenta (PMI ou iPlasma) é TRUE e a coluna is_long_method é FALSE;
- ADCI, quando o valor da coluna correspondente à ferramenta (PMI ou iPlasma) é FALSE e a coluna is_long_method também é FALSE;
- ADII, quando o valor da coluna correspondente à ferramenta (PMI ou iPlasma) é FALSE e a coluna is_long_method é TRUE.

Reply:
that's not it. a questao e' q PMD ou iPlasma dar false ou true nao e' "isPMD" ou "isiPlasma" - e' mais "PMD_is_long_method" e "iPlasma_is_long_method", e isso deixaria mais claro aqui o que se está de facto a fazer. o "isIs_long_method" também deveria ser refactored para algo como "true_is_long_method"

Reply:
isPMD, pelo nome, é um boolean que retorna se algo usou esta ferramenta ou não. O mesmo para isIPlasma. 
Não é isto que está a ser feito ou analisado. 
O que está de facto a acontecer é que na coluna correspondente à ferramenta, é retirado o booleano que indica se a ferramenta considerou aquele método "is_long_method" ou não. O nome tem de refletir isto. 

@ftfni-iscteiul @lrnsa-iscteiul @jmalo1-iscteiul @Hugo-Barroca opinions?

Reply:
Nisso eu concordo com a Vanda, acho que vale um refactor. E é bem fácil, é só selecionar o método no ficheiro do model, botão direito e refactor, e isso já mexe no código todo.
Pq realmente, do jeito que está, parece que retorna se algo é PMD, ou se algo é iPlasma. O que não é o caso.
Eu sugeriria algo na linha `isDefectDetectedPMD()`, ou `isLongMethodPMD()`, etc.

Reply:
get_PMD_result e get_iPlasma_result

Reply:
Lino's suggestion seems good to me.

Reply:
mas sem "_" -> isto é java. logo getPMDResult e getIPlasmaResult

Reply:
Perfeito. done :D

Reply:
please push changes